pkg/gardenlet/controller/controllerinstallation/controllerinstallation/reconciler_test.go

```diff
@@ -0,0 +1,147 @@
+			daemonSet   *appsv1.DaemonSet
+			job         *batchv1.Job
+			cronJob     *batchv1.CronJob
+		)
+
+		BeforeEach(func() {
+			pod = &corev1.Pod{}
+			deployment = &appsv1.Deployment{}
+			statefulSet = &appsv1.StatefulSet{}
+			daemonSet = &appsv1.DaemonSet{}
+			job = &batchv1.Job{}
+			cronJob = &batchv1.CronJob{}
+		})
+
+		It("should not change objects if not bootstrapping control plane node", func() {
+			reconciler.BootstrapControlPlaneNode = false
+
+			p := pod.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(p)).To(Succeed())
+			Expect(p).To(Equal(pod))
+
+			d := deployment.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(d)).To(Succeed())
+			Expect(d).To(Equal(deployment))
+
+			s := statefulSet.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(s)).To(Succeed())
+			Expect(s).To(Equal(statefulSet))
+
+			ds := daemonSet.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(ds)).To(Succeed())
+			Expect(ds).To(Equal(daemonSet))
+
+			j := job.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(j)).To(Succeed())
+			Expect(j).To(Equal(job))
+
+			c := cronJob.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(c)).To(Succeed())
+			Expect(c).To(Equal(cronJob))
+		})
+
+		It("should adapt objects if bootstrapping control plane node", func() {
+			reconciler.BootstrapControlPlaneNode = true
+
+			checkHostNetworkAndTolerations := func(podSpec *corev1.PodSpec) {
+				Expect(podSpec.HostNetwork).To(BeTrue())
+				Expect(podSpec.Tolerations).To(Equal([]corev1.Toleration{
+					{Operator: corev1.TolerationOpExists, Effect: corev1.TaintEffectNoSchedule},
+					{Operator: corev1.TolerationOpExists, Effect: corev1.TaintEffectNoExecute},
+				}))
+			}
+
+			for _, obj := range []struct {
+				object    runtime.Object
+				checkFunc func()
+			}{
+				{pod, func() {
+					checkHostNetworkAndTolerations(&pod.Spec)
+				}},
+				{deployment, func() {
+					Expect(deployment.Spec.Replicas).To(Equal(ptr.To(int32(1))))
+					checkHostNetworkAndTolerations(&deployment.Spec.Template.Spec)
+				}},
+				{statefulSet, func() {
+					checkHostNetworkAndTolerations(&statefulSet.Spec.Template.Spec)
+				}},
+				{daemonSet, func() {
+					checkHostNetworkAndTolerations(&daemonSet.Spec.Template.Spec)
+				}},
+				{job, func() {
+					checkHostNetworkAndTolerations(&job.Spec.Template.Spec)
+				}},
+				{cronJob, func() {
+					checkHostNetworkAndTolerations(&cronJob.Spec.JobTemplate.Spec.Template.Spec)
+				}},
+			} {
+				Expect(reconciler.BootstrapControlPlaneNodeFunc(obj.object)).To(Succeed())
+				obj.checkFunc()
+			}
+		})
+	})
+
+	Describe("#CalculateNextUsablePorts", func() {
+		It("should calculate usable ports range", func() {
+			ports, err := reconciler.CalculateNextUsablePorts()
+			Expect(err).To(Succeed())
+			Expect(ports).To(HaveLen(5))
+			allocatedPorts := map[int]struct{}{}
+			for i, p := range ports {
+				ExpectWithOffset(i, p).To(BeNumerically(">", 0))
```

Comment:
Why are you adding an offset here? The assertions are not performed in a helper function or something similar.